### ISSUE TITLE
Menus: Update project structure/builds to include JSON documents from extensions

### ISSUE THREAD

**Issue by lyonsil** [paranext/paranext-core#421]
As a Platform.Bible extension developer, I want to be able to add menus to my webviews and contribute to extensible points in platform-provided menus.

----

Extensions will provide JSON documents that Platform.Bible combines with other JSON documents to produce a final definition of menus to display.  This ticket is about creating the technical contribution mechanism.  Update the build process for extensions including the core extensions to menu JSON according to the menu design.  More details about the overall design are in the [menu contribution design document](https://docs.google.com/document/d/1N9BVj5w1b3H90Sjjc1lRN2zbalL-beDHmh4jxxExtiw/edit?usp=drive_link).

This issue is not about validating the documents or serving the documents; it is simply about including them in the extensions.

For this task, "done" should include updating the extensions that are part of the "core" repo to provide menu JSON documents. 

Testing Idea: test to make sure the json document makes it into the final zip. Note, this will be easier to test when menus end to end are working.

----

<details><summary>Previous details that may be useful for future work</summary>
<p>

There are at least a couple of ways this could work.  The easier, less flexible approach would be for extensions to specify one or more JSON files in their manifest that the platform loads and uses.  The more complex approach would be to update the extension activation process and allow extensions to provide a dynamically generated JSON document.

I think it makes sense to go with the easier approach as menus probably don't need to be (and maybe shouldn't be) so dynamic they can't be described and provided as a static asset.  If someone disagrees, we can update this ticket before it goes through a sizing exercise.  Even the more complex approach probably isn't CRAZY complicated, but it just seems to cause incremental, extra effort without compelling use cases.

</p>
</details>

**Comment by tjcouch-sil** [comment]
Extensions should provide static json files for menus so that Platform.Bible can display menus without having to activate all extensions. If we think of needs for dynamic menus, we can work on that another time.

**Comment by lyonsil** [comment]
I think the core platform's JSON document probably needs to be dynamic (assuming we will have menu items for each window and/or open project), but for extensions I agree the case is less compelling.

**Comment by roopa0222** [comment]
Repo steps.
In my test extension at location C:\Repos\paranext-ext-testing

1. I ran npm run build on terminal
2. Confirmed that menus.json is present in dist/contribution folder
3. I ran npm run package on terminal
4. Confirmed that paranext-extension-template_0.0.1.zip is build in release folder
5. Confirmed that menus.json is present in paranext-extension-template_0.0.1.zip/contribution folder
![Screenshot 2024-04-24 145220](https://github.com/paranext/paranext-core/assets/133702629/31114bab-b44b-4194-8673-31ddd300a471)
![Screenshot 2024-04-24 145131](https://github.com/paranext/paranext-core/assets/133702629/04c5ca14-880b-4a51-b645-db128f33b884)
